ddtrace/contrib/internal/aiokafka/patch.py

```diff
@@ -0,0 +1,201 @@
+from ddtrace.internal.schema.span_attribute_schema import SpanDirection
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.utils import set_argument_value
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.internal.utils.wrappers import unwrap as _u
+from ddtrace.propagation.http import HTTPPropagator
+
+
+config._add(
+    "aiokafka",
+    dict(
+        _default_service=schematize_service_name("kafka"),
+        distributed_tracing_enabled=asbool(os.getenv("DD_KAFKA_PROPAGATION_ENABLED", default=False)),
+    ),
+)
+
+
+def get_version() -> str:
+    return getattr(aiokafka, "__version__", "")
+
+
+def _supported_versions() -> Dict[str, str]:
+    return {"aiokafka": ">=0.9.0"}
+
+
+def common_aiokafka_tags(topic, bootstrap_servers):
+    return {
+        COMPONENT: config.aiokafka.integration_name,
+        TOPIC: topic,
+        MESSAGING_DESTINATION_NAME: topic,
+        MESSAGING_SYSTEM: SERVICE,
+        HOST_LIST: bootstrap_servers,
+    }
+
+
+def common_consume_aiokafka_tags(topic, bootstrap_servers, group_id):
+    tags = common_aiokafka_tags(topic, bootstrap_servers)
+    tags.update(
+        {
+            SPAN_KIND: SpanKind.CONSUMER,
+            GROUP_ID: group_id,
+        }
+    )
+    return tags
+
+
+def parse_send(instance, args, kwargs):
+    topic = get_argument_value(args, kwargs, 0, "topic")
+    value = get_argument_value(args, kwargs, 1, "value", True)
+    key = get_argument_value(args, kwargs, 2, "key", True) or None
+    partition = get_argument_value(args, kwargs, 3, "partition", True)
+    headers = get_argument_value(args, kwargs, 5, "headers", True) or []
+    servers = instance.client._bootstrap_servers
+
+    return topic, value, headers, partition, key, servers
+
+
+async def traced_send(func, instance, args, kwargs):
+    topic, value, headers, partition, key, bootstrap_servers = parse_send(instance, args, kwargs)
+
+    with core.context_with_data(
+        "aiokafka.send",
+        span_name=schematize_messaging_operation(PRODUCE, provider="kafka", direction=SpanDirection.OUTBOUND),
+        span_type=SpanTypes.WORKER,
+        service=trace_utils.ext_service(None, config.aiokafka),
+        tags=common_aiokafka_tags(topic, bootstrap_servers),
+    ) as ctx:
+        span = ctx.span
+        core.dispatch("aiokafka.send.start", (topic, value, key, headers, span, partition))
+        args, kwargs = set_argument_value(args, kwargs, 5, "headers", headers, override_unset=True)
+
+        try:
+            result = await func(*args, **kwargs)
+        except BaseException as e:
+            span.set_exc_info(type(e), e, e.__traceback__)
+            span.finish()
+            raise e
+
+        def sent_callback(future):
+            try:
+                result = future.result()
+                core.dispatch("aiokafka.send.completed", (result,))
+            except Exception as e:
+                span.set_exc_info(type(e), e, e.__traceback__)
```

Comment:
Putting this and other `span` method calls into `trace_handlers` would keep this file free of direct uses of the `Span` API, making it easier to instrument without tracing. This may be useful if we want to allow DSM to be used without APM.